### DISCUSSION TITLE
When multiple work items are related to single PR, only one work item is able to pick the associated PR

### DISCUSSION THREAD

**Discussion by harvinders** [rfennell/AzurePipelines#1055] [General]
I have a following template, that iterates through the work items associated with the release and then finds the associated PR for the work item.

I found that if there are multiple work items that were closed by a single PR, then only one work Item displays the PR using the template below. 

Am I using the template correctly? Is there a way around?

```
{{#forEach this.workItems}}
{{#if isFirst}}### BUGS {{/if}}
{{#if (contains (lookup this.fields 'System.WorkItemType') 'Bug') }}
*  **[{{this.id}}]({{replace this.url "_apis/wit/workItems" "_workitems/edit"}})** - {{lookup this.fields 'System.Title'}}
{{#forEach this.relations}}
{{#if (contains this.attributes.name 'Pull Request')}}
{{#with (lookup_a_pullrequest ../../pullRequests  this.url)}}
    - {{this.title}} (**[PR {{this.pullRequestId}}]({{replace (replace this.url "_apis/git/repositories" "_git") "pullRequests" "pullRequest"}})**)
{{/with}}
{{/if}}
{{/forEach}}
{{/if}}
{{/forEach}}
```

**Comment by rfennell** [comment]
It is down to how you ended your PR, a merge commit, a squash, they all look a little different. So I am not surprised by the results you see.

Have you tried looking at that `PullRquests` array, this comes from the other direction. Might help
